blog/03-08-2023-Build-A-Full-Stack-Decentralized-Fund-Raising-Dapp-On-The-Celo-Blockchain/Build-A-Full-Stack-Decentralized-Fund-Raising-Dapp-On-The-Celo-Blockchain.md

```diff
@@ -0,0 +1,666 @@
+
+
+
+```javascript
+ const addCampaign = async (_image, _description, _beneficiary, _goal) => {
+    try {
+      await contract.methods
+        .createCampaign(_image, _description, _beneficiary, _goal)
+        .send({ from: address });
+      getCampaigns();
+    } catch (error) {
+      alert(error);
+    }
+  };
+```
+After that, we have the `addCampaign` function, which takes in the required parameters to create a new campaign and sends a transaction using the `createCampaign` method of the contract object. It then calls the `getCampaigns` function to update the list of campaigns.
+
+```javascript
+ const fundCampaign = async (_index, _ammount) => {
+    try {
+      const cUSDContract = new kit.web3.eth.Contract(IERC, cUSDContractAddress);
+
+      await cUSDContract.methods
+        .approve(contractAddress, _ammount)
+        .send({ from: address });
+      await contract.methods
+        .fundCampaign(_index, _ammount)
+        .send({ from: address });
+      getCampaigns();
+      getBalance();
+      alert("you have successfully sent cusd to this user");
+    } catch (error) {
+      alert(error);
+    }
+  };
+```
+Lastly, we have the `fundCampaign` function which takes in the index of the campaign and the amount of cUSD to be sent. It first approves the contract to spend the specified amount of cUSD using the approve method of the `cUSDContract` object. It then sends a transaction to the contract to fund the `campaign` using the `fundCampaign` method. Finally, it calls the `getCampaigns` and `getBalance` functions to update the list of `campaigns` and the user's `cUSD` balance.
+
+```javascript
+useEffect(() => {
+connectToWallet();
+}, []);
+
+useEffect(() => {
+if (kit && address) {
+getBalance();
+}
+}, [kit, address, getBalance]);
+
+useEffect(() => {
+if (contract) {
+getCampaigns();
+}
+}, [contract, getCampaigns]);
+
+```
+We use useEffect to call the connectToWallet, getBalance, and getCampaigns functions at the appropriate times.
+
+```javascript
+return (
+    <div className="App">
+      <Home cUSDBalance={cUSDBalance} addCampaign={addCampaign} />
+      <Campaigns
+        campaigns={campaigns}
+        fundCampaign={fundCampaign}
+        walletAddress={address}
+      />
+    </div>
+  );
+}
+
+export default App;
+
+```
+And finally, we render the App component and return the Home and Campaigns components with the necessary props.
+
+## Next Steps
+
+I hope you learned a lot from this tutorial. Here are some relevant links that would aid your learning further.
+
+- <https://docs.celo.org/>
+- <https://solidity-by-example.org/>
+- <https://docs.soliditylang.org/en/v0.8.17/>
+
+## About the author
+
+I'm Jonathan Iheme, A full stack block-chain Developer from Nigeria.
```

Comment:
not compulsory but why not add a link to you linkedIn Page, helps with the traffic
something like, You can connect with me on [LinkedIn](www.you.link.here)